tidy3d/components/data/dataset.py

```diff
@@ -337,6 +337,15 @@ def sel_mode_index(self, mode_index: pd.NonNegativeInt) -> FieldDataset:
 
         return self.copy(update=fields)
 
+    def sel_freq_index(self, freq_index: pd.NonNegativeInt) -> FieldDataset:
+        """Return :class:`.FieldData` for the specificed frequency index."""
+
+        fields = {}
+        for field_name, data in self.field_components.items():
+            fields[field_name] = data.isel(f=[freq_index])
```

Comment:
I think `freq_index` doesn't need to be in a list, could be wrong.

Reply:
I get caught by this every now and then, basically the difference is whether the isel-ed dim will be squeezed out or not, so if you want it to stay, you put it in a list.

Reply:
actually, would it make sense to make it to accept any of `freq_index`, `[freq_index]`, `[freq1, freq2, ...]` which would be directly passed to `data.isel`, so there is a choice depending on what's needed? And same for `sel_mode_index`?

Reply:
Well, now that you bring it up, I feel like we should just wrap `.isel` in the base `AbstractFieldDataset` class:

```
def isel(self, **isel_kwargs) -> AbstractFieldDataset:
    return {key: val.isel(**isel_kwargs) for key, val in self.field_components.items()}
```

We've discussed this a few times with @tylerflex. I think one argument against is that we just don't want to end up wrapping all sorts of xarray functions. But given that we are already wrapping `isel` effectively twice, once for mode index and once for frequency index, it seems like it makes more sense to just unify this.

My main objection is actually this: when you call `isel` on a `FieldData`, the returned `FieldData` will then have a mismatch in e.g. `FieldData.monitor.freqs` and `FieldData.Ex.f`. This is an issue with the current implementation of `sel_freq_index` too. How should we handle this?

1. Wrap `isel` and let the user worry about it
2. Keep the existing functions but also define `AbstractFieldData.sel_freq_index` to change the list of frequencies in `self.monitor` before calling `super().sel_freq_index`.
3. Remove the existing isel wrappers altogether and just do it manually when needed (it's basically one line of code if you write it as dict comprehension as above...)

Having written all this up I'm kinda leaning towards 3.